turnkey-version.py

```diff
@@ -1,52 +1,54 @@
-#!/usr/bin/python
+#! /usr/bin/python
 # Ad-hoc version detection for a system installed from a TurnKey
 # appliance. This is a transitory package so we can identify the versions
 # in current appliance which were designed before we needed this for
 # tklbam. In the future it shouldn't be needed as all appliances will be
 # marked with their versions.
 
+from __future__ import print_function
+from builtins import str
 from os.path import join
 
 import sys
 
 def usage(e=None):
     if e:
-        print >> sys.stderr, "error: " + str(e)
+        print(u'error: ' + str(e), file=sys.stderr)
 
-    print >> sys.stderr, "Syntax: %s [rootfs]" % sys.argv[0]
+    print(u"Syntax: %s [rootfs]" % sys.argv[0], file=sys.stderr)
```

Comment:
No need for `u` strings, surely? :)

Reply:
From http://python-future.org/compatible_idioms.html

> Unicode (text) string literals
> If you are upgrading an existing Python 2 codebase, it may be preferable to mark up all string literals as unicode explicitly with u prefixes:
> 
> 
> # Python 2 only
> s1 = 'The Zen of Python'
> s2 = u'きたないのよりきれいな方がいい\n'
> 
> # Python 2 and 3
> s1 = u'The Zen of Python'
> s2 = u'きたないのよりきれいな方がいい\n'
> The futurize and python-modernize tools do not currently offer an option to do this automatically.
> 
> If you are writing code for a new project or new codebase, you can use this idiom to make all string literals in a module unicode strings:
> 
> # Python 2 and 3
> from __future__ import unicode_literals    # at top of module
> 
> s1 = 'The Zen of Python'
> s2 = 'きたないのよりきれいな方がいい\n'
> See http://python-future.org/unicode_literals.html for more discussion on which style to use.

While Python2 automatically switches unicode strings and byte-code strings, Python3 does not.  The `join` statement was throwing an error in Python3 until I changed the string to unicode.  I changed several other strings just for consistency.  Note: I could have used the second method by importing `unicode_literals`.

For me, this was primarily an exercise to learn more about writing Python 2/3 compatible code.  I chose `turnkey-version.py` because it was a single file that did not call any of the utilities in `turnkey-pylib`.

The real question is do we need an interim version that supports both Python 2 & 3.  I think Jeremy is leaning toward a clean break moving to Python 3 only, but this will mean older 14.2 appliances won't be able to upgrade.

Reply:
AFAICT just use `.format()` instead of `%s` here with regular ol non-`u` strings and everything magically fixes itself. :)

Reply:
Yeah `.format()` FTW!

I.e. something like this:
```
print("Syntax: {} [rootfs]".format(sys.argv[0]), file=sys.stderr)
```

Reply:
[somewhat OT] Wow! I just discovered "fstrings" in python3.6+.

It's another option which has all the advantages of `.format()`, plus is faster that the old `%` formatting (using `.format()` is slower than `%`) and I would argue is also easier to read than either! Obviously won't be of any use to use until Buster, but cool none the less.

E.g.:
```
a = 123
b = 456
print(f"{a} + {b} = {a+b}")
```
Outputs:
```
123 + 456 = 579
```

Reply:
Just my opinion but I find f-strings really ugly and far too magic in that they suck in variables from the outside environment.. plus the ability to do maths like that is too limited for real world usage IMHO and mostly makes for "neat" examples. Alas.

Reply:
Ah ok. I've never used them in the real world, but as a newish python recruit (with a historic love of bash) I like the "readability" of them (i.e. more like a bash string with variable substitution). I can certainly see that the maths side of things probably has little real world application though.